File: Java/UPCMatricula/src/controllers/MatriculaController.java
package controllers;

import models.Alumno;
import models.Curso;

public class MatriculaController {
    private Curso curso;
    
    public MatriculaController() {
        this.crearCurso();
    }
    
    private void crearCurso() {
        this.curso = new Curso("POO");
    }
    
    public Alumno crearAlumno(String codigo, String nombre, String apellido) {
        Alumno alumno = new Alumno(codigo, nombre, apellido);
        return alumno;
    }

    public String matricular(String codigo, String nombre, String apellido) {
        Alumno alumno = this.crearAlumno(codigo, nombre, apellido);
        this.curso.agregarAlumno(alumno);
        return "Alumno matriculado";
    }
    
    public String reporteMatricula() {
        return this.curso.listarMatriculados();
    }
}

File: Ruby/UPCMatricula/controllers/MatriculaController.rb
require_relative '../models/Alumno'
require_relative '../models/Curso'

class MatriculaController

    def initialize
        @curso = self.crear_curso("POO")
    end

    def crear_curso(nombre)
        return Curso.new(nombre)
    end

    def crear_alumno(codigo, nombre, apellido)
        return Alumno.new(codigo, nombre, apellido)
    end

    def matricular(codigo, nombre, apellido)
        alumno = self.crear_alumno(codigo, nombre, apellido)
        @curso.agregar_alumno(alumno)
        return "Alumno matriculado"
    end

    def reporte_matricula
        return @curso.listar_matriculados()
    end
    
end
